Null Pointer Exception in NvdCveUpdater

### Reporting Bugs/Errors
Hi all, this the first time I am using your tool. While downloading the databases for the first time, I get a `NullPointerException` error in the `NvdCveUpdater`. I tried your command line version and also the `Gradle` version. I will appreciate if you can provide any feedback regarding this error. More details below:

```
[INFO] Checking for updates
[INFO] NVD CVE requires several updates; this could take a couple of minutes.
[INFO] Download Started for NVD CVE - 2002
[INFO] Download Started for NVD CVE - 2003
[INFO] Download Complete for NVD CVE - 2003  (563 ms)
[INFO] Download Started for NVD CVE - 2004
[INFO] Processing Started for NVD CVE - 2003
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.fasterxml.jackson.module.afterburner.util.MyClassLoader (file:/home/leob/Desktop/Tools/DependencyChecker/lib/jackson-module-afterburner-2.12.2.jar) to method java.lang.ClassLoader.findLoadedClass(java.lang.String)
WARNING: Please consider reporting this to the maintainers of com.fasterxml.jackson.module.afterburner.util.MyClassLoader
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[INFO] Download Complete for NVD CVE - 2002  (817 ms)
[INFO] Download Started for NVD CVE - 2005
[INFO] Download Complete for NVD CVE - 2004  (489 ms)
[INFO] Download Started for NVD CVE - 2006
[INFO] Download Complete for NVD CVE - 2005  (747 ms)
[INFO] Download Started for NVD CVE - 2007
[INFO] Download Complete for NVD CVE - 2006  (937 ms)
[INFO] Download Started for NVD CVE - 2008
[INFO] Download Complete for NVD CVE - 2007  (921 ms)
[INFO] Download Started for NVD CVE - 2009
[INFO] Processing Started for NVD CVE - 2002
[INFO] Processing Started for NVD CVE - 2004
[INFO] Processing Started for NVD CVE - 2005
[INFO] Download Complete for NVD CVE - 2008  (1047 ms)
[INFO] Download Started for NVD CVE - 2010
[INFO] Download Complete for NVD CVE - 2009  (963 ms)
[INFO] Download Started for NVD CVE - 2011
[INFO] Processing Started for NVD CVE - 2006
[INFO] Processing Started for NVD CVE - 2007
[INFO] Download Complete for NVD CVE - 2010  (947 ms)
[INFO] Download Started for NVD CVE - 2012
[INFO] Processing Started for NVD CVE - 2008
[INFO] Processing Started for NVD CVE - 2009
[INFO] Download Complete for NVD CVE - 2011  (837 ms)
[INFO] Download Started for NVD CVE - 2013
[INFO] Processing Started for NVD CVE - 2010
[INFO] Download Complete for NVD CVE - 2012  (1073 ms)
[INFO] Download Started for NVD CVE - 2014
[INFO] Download Complete for NVD CVE - 2013  (1327 ms)
[INFO] Download Started for NVD CVE - 2015
[INFO] Processing Started for NVD CVE - 2011
[INFO] Processing Started for NVD CVE - 2012
[INFO] Processing Started for NVD CVE - 2013
[INFO] Download Complete for NVD CVE - 2014  (1305 ms)
[INFO] Download Started for NVD CVE - 2016
[INFO] Processing Started for NVD CVE - 2014
[INFO] Download Complete for NVD CVE - 2015  (957 ms)
[INFO] Download Started for NVD CVE - 2017
[INFO] Processing Started for NVD CVE - 2015
[INFO] Download Complete for NVD CVE - 2016  (1236 ms)
[INFO] Processing Started for NVD CVE - 2016
[INFO] Download Started for NVD CVE - 2018
[INFO] Download Complete for NVD CVE - 2017  (1705 ms)
[INFO] Processing Started for NVD CVE - 2017
[INFO] Download Started for NVD CVE - 2019
[INFO] Download Complete for NVD CVE - 2018  (1645 ms)
[INFO] Processing Started for NVD CVE - 2018
[INFO] Download Started for NVD CVE - 2020
[INFO] Download Complete for NVD CVE - 2019  (2151 ms)
[INFO] Processing Started for NVD CVE - 2019
[INFO] Download Started for NVD CVE - 2021
[INFO] Download Complete for NVD CVE - 2021  (747 ms)
[INFO] Processing Started for NVD CVE - 2021
[INFO] Download Complete for NVD CVE - 2020  (2378 ms)
[INFO] Processing Started for NVD CVE - 2020
[ERROR] java.util.concurrent.ExecutionException: java.lang.NullPointerException
org.owasp.dependencycheck.data.update.exception.UpdateException: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at org.owasp.dependencycheck.data.update.NvdCveUpdater.performUpdate(NvdCveUpdater.java:298)
	at org.owasp.dependencycheck.data.update.NvdCveUpdater.update(NvdCveUpdater.java:125)
	at org.owasp.dependencycheck.Engine.doUpdates(Engine.java:860)
	at org.owasp.dependencycheck.Engine.initializeAndUpdateDatabase(Engine.java:667)
	at org.owasp.dependencycheck.Engine.analyzeDependencies(Engine.java:593)
	at org.owasp.dependencycheck.App.runScan(App.java:254)
	at org.owasp.dependencycheck.App.run(App.java:186)
	at org.owasp.dependencycheck.App.main(App.java:81)
Caused by: java.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at org.owasp.dependencycheck.data.update.NvdCveUpdater.performUpdate(NvdCveUpdater.java:288)
	... 7 common frames omitted
Caused by: java.lang.NullPointerException: null
	at org.owasp.dependencycheck.data.nvd.ecosystem.UrlEcosystemMapper.getEcosystem(UrlEcosystemMapper.java:68)
	at org.owasp.dependencycheck.data.nvd.ecosystem.CveEcosystemMapper.getEcosystem(CveEcosystemMapper.java:74)
	at org.owasp.dependencycheck.data.update.nvd.NvdCveParser.parse(NvdCveParser.java:97)
	at org.owasp.dependencycheck.data.update.nvd.ProcessTask.importJSON(ProcessTask.java:139)
	at org.owasp.dependencycheck.data.update.nvd.ProcessTask.processFiles(ProcessTask.java:152)
	at org.owasp.dependencycheck.data.update.nvd.ProcessTask.call(ProcessTask.java:113)
	at org.owasp.dependencycheck.data.update.nvd.ProcessTask.call(ProcessTask.java:40)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
[INFO] Skipping RetireJS update since last update was within 24 hours.
[WARN] Unable to update 1 or more Cached Web DataSource, using local data instead. Results may not include recent vulnerabilities.
[ERROR] Unable to continue dependency-check analysis.
[ERROR] One or more fatal errors occurred
[ERROR] java.util.concurrent.ExecutionException: java.lang.NullPointerException
[ERROR] No documents exist
```

--- a/core/src/main/java/org/owasp/dependencycheck/data/nvd/ecosystem/UrlEcosystemMapper.java
+++ b/core/src/main/java/org/owasp/dependencycheck/data/nvd/ecosystem/UrlEcosystemMapper.java
@@ -17,12 +17,16 @@
  */
 package org.owasp.dependencycheck.data.nvd.ecosystem;
 
+import java.util.Objects;
+import java.util.Optional;
 import java.util.TreeMap;
 
 import javax.annotation.concurrent.NotThreadSafe;
 
+import org.owasp.dependencycheck.data.nvd.json.CVEJSON40Min11;
 import org.owasp.dependencycheck.data.nvd.json.DefCveItem;
 import org.owasp.dependencycheck.data.nvd.json.Reference;
+import org.owasp.dependencycheck.data.nvd.json.References;
 
 import com.hankcs.algorithm.AhoCorasickDoubleArrayTrie;
 import com.hankcs.algorithm.AhoCorasickDoubleArrayTrie.Hit;
@@ -65,8 +69,13 @@ public class UrlEcosystemMapper {
      * @return the ecosystem
      */
     public String getEcosystem(DefCveItem cve) {
-        if (cve.getCve().getReferences() != null) {
-            for (Reference r : cve.getCve().getReferences().getReferenceData()) {
+        References references = Optional.ofNullable(cve)
+                .map(DefCveItem::getCve)
+                .map(CVEJSON40Min11::getReferences)
+                .orElse(null);
+
+        if (Objects.nonNull(references)) {
+            for (Reference r : references.getReferenceData()) {
 
                 final Hit<String> ecosystem = search.findFirst(r.getUrl());
                 if (ecosystem != null) {

--- a/core/src/test/java/org/owasp/dependencycheck/data/nvd/ecosystem/UrlEcosystemMapperTest.java
+++ b/core/src/test/java/org/owasp/dependencycheck/data/nvd/ecosystem/UrlEcosystemMapperTest.java
@@ -1,6 +1,7 @@
 package org.owasp.dependencycheck.data.nvd.ecosystem;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
 
 import org.junit.Test;
 import org.owasp.dependencycheck.analyzer.PythonPackageAnalyzer;
@@ -35,5 +36,22 @@ public class UrlEcosystemMapperTest {
         defCveItem.setCve(cve);
         
         return defCveItem;
+    }
+
+    @Test
+    public void testGetEcosystemMustHandleNullCveReferences() {
+        // Given
+        UrlEcosystemMapper mapper = new UrlEcosystemMapper();
+
+        CVEJSON40Min11 cve = new CVEJSON40Min11();
+
+        DefCveItem cveItem = new DefCveItem();
+        cveItem.setCve(cve);
+
+        // When
+        String output = mapper.getEcosystem(cveItem);
+
+        // Then
+        assertNull(output);
     }
 }

--- a/core/src/test/java/org/owasp/dependencycheck/data/nvd/ecosystem/UrlEcosystemMapperTest.java
+++ b/core/src/test/java/org/owasp/dependencycheck/data/nvd/ecosystem/UrlEcosystemMapperTest.java
@@ -54,4 +54,30 @@ public class UrlEcosystemMapperTest {
         // Then
         assertNull(output);
     }
+
+    @Test
+    public void testGetEcosystemMustHandleNullCve() {
+        // Given
+        UrlEcosystemMapper mapper = new UrlEcosystemMapper();
+
+        DefCveItem cveItem = new DefCveItem();
+
+        // When
+        String output = mapper.getEcosystem(cveItem);
+
+        // Then
+        assertNull(output);
+    }
+
+    @Test
+    public void testGetEcosystemMustHandleNullCveItem() {
+        // Given
+        UrlEcosystemMapper mapper = new UrlEcosystemMapper();
+
+        // When
+        String output = mapper.getEcosystem(null);
+
+        // Then
+        assertNull(output);
+    }
 }

